pkg/hooks/manager.go

```diff
@@ -260,75 +279,152 @@ func (m *hookManager) onDefineDomainCallback(callback *callBackClient, domainSpe
 }
 
 func (m *hookManager) PreCloudInitIso(vmi *v1.VirtualMachineInstance, cloudInitData *cloudinit.CloudInitData) (*cloudinit.CloudInitData, error) {
-	if callbacks, found := m.CallbacksPerHookPoint[hooksInfo.PreCloudInitIsoHookPointName]; found {
-		for _, callback := range callbacks {
-			if callback.Version == hooksV1alpha2.Version {
-				var resultData *cloudinit.CloudInitData
-				vmiJSON, err := json.Marshal(vmi)
-				if err != nil {
-					return cloudInitData, fmt.Errorf("failed to marshal VMI spec: %v, err: %v", vmi, err)
-				}
+	dataToJSON := func() ([3][]byte, error) {
```

Comment:
For readability sake, is it possible to move this func definition to its own separate helper function?

Reply:
Fixed